src/networking.c

```diff
@@ -2155,7 +2155,7 @@ void readQueryFromClient(connection *conn) {
 
     qblen = sdslen(c->querybuf);
     if (c->querybuf_peak < qblen) c->querybuf_peak = qblen;
-    c->querybuf = sdsMakeRoomFor(c->querybuf, readlen);
+    c->querybuf = sdsMakeRoomForExact(c->querybuf, readlen);
```

Comment:
i discussed this in length with @yoav-steinberg last week.
we think that in some cases we still want to use the greedy version here.

there are some 3 states here:
1. we're allocating the query buffer for the first time, in order to read a small argument into it. in this case we wanna use the non-greedy mode.
2. we're accumulating again and again to the query buffer, in this case we wanna use the greedy mode.
3. we detected a BIG_ARG, and we're gonna allocate a dedicated query buffer for it (we should set a boolean flag in the `if` above), in which case we wanna use the non-greedy mode.

Reply:
Thanks @oranagra, It's hard to come up with a good name and comment.